Updated SharePoint Reference (#1006)
mulesoft/docs-connectors: sharepoint/3.1/modules/ROOT/pages/sharepoint-connector-reference.adoc

== Configurations
---
[[sharepoint-online]]
=== SharePoint Online

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
|Name | String | The name for this configuration. Connectors reference the configuration with this name. | | x
| Connection a| * <<sharepoint-online_okta, Okta>>
* <<sharepoint-online_online, Online>>
* <<sharepoint-online_security-token, Security Token>>
 | The connection types to provide to this configuration. | | x
| Charset a| Enumeration, one of:

** US_ASCII
** ISO_8859_1
** UTF_8
** UTF_16BE
** UTF_16LE
** UTF_16 |  |  UTF_8 |
| Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration. This does not mean that the platform expires the instance at the exact moment that it becomes eligible. Mule purges the instances as appropriate. |  |
|===

==== Connection Types
[[sharepoint-online_okta]]
===== Okta

====== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required

> | Okta Username a| String |  Username to use to log in to the Okta Identity Provider |  | x

Reviewer: Not super important but there is a mix of period and no period in the table. I think , in this case, they all need periods to follow the consistency rule.

Author: Not fixing these because they are all over and not that important.

| Okta Password a| String |  Password to use to log in to the Okta Identify Provider |  | x
| Okta Domain a| String |  Domain name for the Okta Identity Provider |  | x
| Okta API token a| String |  |  | x
| Sharepoint Online embedded link from Okta a| String |  |  | x
| Disable SSL certificate validation a| Boolean |  Disables the CN (Common Name) check so that the server won't throw an exception if the HTTPS certificate is not signed by a trusted partner.

NOTE: Disabling the CN check is not recommended for production environments. |  false |
| Host a| String |  Hostname for the proxy server. |  |
| Port a| Number |  Port number for the proxy server. |  |
| Username a| String |  Username to use for authenticating against the proxy server |  |
| Password a| String |  Password to use for authenticating against the proxy server |  |

> | NTLM Domain a| String |  |  |

Reviewer 2: The description here is missing. It should be: Domain to use for authenticating against the proxy server

> | NTLM Domain a| String |  Domain name to use for NTLM authentication|  |

Reviewer 2: description here should be: Domain to use for authenticating against the proxy server

| Online Username a| String |  Username to log in with. Do not include the domain name. |  | x
| Online Password a| String |  Password to log in with |  | x
| Site Url a| String |  URL for the target SharePoint instance |  | x
| Disable SSL certificate validation a| Boolean |  Disables the CN (Common Name) check so that the server won't throw an exception if the HTTPS certificate is not signed by a trusted partner.

NOTE: Disabling the CN check is not recommended for production environments.  |  false |
| Host a| String |  Hostname for the proxy server. |  |
| Port a| Number |  Port number for the proxy server. |  |
| Username a| String |  Username to use for authenticating against the proxy server |  |
| Password a| String |  Password to use for authenticating against the proxy server |  |
| NTLM Domain a| String |  Domain to use for authenticating against the proxy server |  |
| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  |
|===
[[sharepoint-online_security-token]]
===== Security Token

> Deprecated the SecurityTokenConnectionStrategy Connection Provider because the BasicConnectionProvider already obtains the security token

Reviewer: ```suggestion
The Security Token connection provider is deprecated because the Basic Connection provider already obtains the security token.
```

| Security Token a| String |  |  | x
| Security Token Provider a| <<SecurityTokenProvider>> |  (Optional) Provider instance that provides the security token when required. |  |
| Site Url a| String |  URL for the target SharePoint instance |  | x
| Disable SSL certificate validation a| Boolean |  Disables the CN (Common Name) check so that the server won't throw an exception if the HTTPS certificate is not signed by a trusted partner.

NOTE: Disabling the CN check is not recommended for production environments.  |  false |
| Host a| String |  Hostname for the proxy server |  |
| Port a| Number |  Port number for the proxy server |  |
| Username a| String |  Username to use for authenticating against the proxy server |  |
| Password a| String |  Password to use for authenticating against the proxy server |  |
| NTLM Domain a| String |  Domain to use for authenticating against the proxy server |  |
| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  |
|===

[[sharepoint-onpremise]]

> === SharePoint Onpremise

Reviewer: On-premises

| Site Url a| String |  URL for the target SharePoint instance |  | x
| SPN a| String |  (Optional) Service principal name for the SharePoint Web Service |  |
| Realm a| String |  (Optional) The case-sensitive default realm (Domain Name) that the user belongs to. |  |
| KDC a| String |  (Optional) The KDC (usually the Domain Controller name) to authenticate the user. |  |
| Login Properties File Path a| String |  (Optional) Path to a customized login properties file. The default values work for most cases |  |
|  Properties File Path a| String |  (Optional) Path to a customized Kerberos properties file. |  |
| Disable SSL certificate validation a| Boolean |  Disables the CN (Common Name) check so that the server won't throw an exception if the HTTPS certificate is not signed by a trusted partner.

NOTE: Disabling the CN check is not recommended for production environments.  |  false |
| Host a| String |  Hostname for the proxy server. |  |
| Port a| Number |  Port number for the proxy server. |  |
| Username a| String |  Username to use for authenticating against the proxy server |  |
| Password a| String |  Password to use for authenticating against the proxy server |  |

> | NTLM Domain a| String |  Domain to use for authenticating against the proxy server ServerRelativeUrl |  |

Reviewer 2: please remove 'ServerRelativeUrl' part

| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  |
|===
[[sharepoint-onpremise_ntlm]]
===== NTLM

====== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Domain a| String |  Domain name to use for NTLM authorization |  | x
| Ntlm Username a| String |  Username to log in with. Do not include the domain name. |  | x
| Ntlm Password a| String |  Password NTLM authorization. |  | x
| Site Url a| String |  URL for the target SharePoint instance |  | x
| Disable SSL certificate validation a| Boolean |  Disables the CN (Common Name) check so that the server won't throw an exception if the HTTPS certificate is not signed by a trusted partner.

NOTE: Disabling the CN check is not recommended for production environments.  |  false |
| Host a| String |  Hostname for the proxy server. |  |
| Port a| Number |  Port number for the proxy server. |  |
| Username a| String |  Username to use for authenticating against the proxy server |  |

> | Password to use for authenticating against the proxy server  a| String |  Password to use for authenticating against the proxy server |  |

Reviewer 2: first column should be only Password

| NTLM Domain a| String |  Domain to use for authenticating against the proxy server|  |
| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  |
|===
[[sharepoint-onpremise_security-token]]
===== Security Token

> Deprecated the SecurityTokenConnectionStrategy connection provider because the BasicConnectionProvider already obtains the security token

Reviewer: ```suggestion
The Security Token connection strategy connection provider is deprecated because the Basic Connection provider already obtains the security token.
```

====== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Security Token a| String |  |  | x
| Security Token Provider a| <<SecurityTokenProvider>> |  (Optional) Provider instance that provides the security token when required. |  |
| Site Url a| String |  URL for the target SharePoint instance |  | x
| Disable SSL certificate validation a| Boolean |  Disables the CN (Common Name) check so that the server won't throw an exception if the HTTPS certificate is not signed by a trusted partner.

NOTE: Disabling the CN check is not recommended for production environments.  |  false |
| Host a| String |  Hostname for the proxy server. |  |
| Port a| Number |  Port number for the proxy server. |  |
| Username a| String |  Username to use for authenticating against the proxy server. |  |
| Password a| String |  Password to use for authenticating against the proxy server. |  |

> | NTLM Domain a| String |  The domain to authenticate against the proxy. |  |

Reviewer 2: description should be: The domain to use to authenticate against the proxy server

| Since a| String |  Required date format is `yyyy-MM-ddTHH:mm:ssZ`. If left empty, the UTC moment of time is used as a starting point |  |
| Object Type a| Enumeration, one of:

** FOLDER
** FILE
** LIST
** LIST_ITEM |  The type of objects to listen for |  | x
| List Id a| String |  Used when listening for modified list items or files |  |
| Recursive a| Boolean |  Used when listening for modified files or folders |  false |
| Path a| String |  |  / |
| Primary Node Only a| Boolean |  Whether this source should be executed only on the primary node when running in Cluster mode |  |
| Scheduling Strategy a| scheduling-strategy |  Configures the scheduler that triggers the polling |  | x
| Redelivery Policy a| <<RedeliveryPolicy>> |  Defines a policy for processing the redelivery of the same message |  |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

==== Output
[%autowidth.spread]
|===
|Type |Object
| Attributes Type a| Any
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

== Operations

[[attachFile]]
=== Attach File
`<sharepoint:attach-file>`

> Attach a file to an item from a SharePoint list

Reviewer: ```suggestion
Attach a file to an item from a SharePoint list.
```

[[fileAdd]]
=== File Add
`<sharepoint:file-add>`

Add a file

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| File Server Relative Url a| String |  |  | x
| File Content Stream a| Binary |  |  |
| Overwrite a| Boolean |  |  false |
| Target Variable a| String |  The name of a variable that stores the operation's output |  |
| Target Value a| String |  An expression that evaluates against the operation's output. The outcome of this expression is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

==== Output
[%autowidth.spread]
|===
|Type |<<SharepointFile>>
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[fileApprove]]
=== File Approve
`<sharepoint:file-approve>`

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required

> | Configuration | String | The name of the configuration to use. | | x

Reviewer: ```suggestion
| Configuration | String | Name of the configuration to use | | x
```

| File Server Relative Url a| String |  |  | x
| Comment a| String |  |  |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[fileCheckIn]]
=== File Check In
`<sharepoint:file-check-in>`

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required

> | Configuration | String | The name of the configuration to use. | | x

Reviewer: ```suggestion
| Configuration | String | Name of the configuration to use | | x
```

| File Server Relative Url a| String |  |  | x
| Check In Type a| Enumeration, one of:

** MINOR_CHECK_IN
** MAJOR_CHECK_IN
** OVERWRITE_CHECK_IN |  |  | x
| Comment a| String |  |  | x
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[fileCheckOut]]
=== File Check Out
`<sharepoint:file-check-out>`

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x

> | File Server Relative Url a| String |  |  | x

Reviewer: I'll just assume `payload` was intentionally removed because it is no longer the default?

| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[fileCopyTo]]
=== File Copy To
`<sharepoint:file-copy-to>`

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required

> | Configuration | String | The name of the configuration to use. | | x

Reviewer: ```suggestion
| Configuration | String | Name of the configuration to use | | x
```

> | File Server Relative Url a| String |  |  | x

Reviewer: No default?

| New File Server Relative Url a| String |  |  | x
| Overwrite a| Boolean |  |  false |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[fileDelete]]
=== File Delete
`<sharepoint:file-delete>`

Deletes a file.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required

> | Configuration | String | The name of the configuration to use. | | x

Reviewer: This is not important. I was only removing the period for consistency's sake. I'll quit commenting on it though.

=== Folder Delete
`<sharepoint:folder-delete>`

Deletes a folder from a Document list

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| Url a| String |  Server relative URL of the folder to delete. For example, `/Shared Documents/new folder`, deletes `new folder` from the `Shared Documents` list. |  | x
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[folderQuery]]
=== Folder Query
`<sharepoint:folder-query>`

Retrieves all folders that matches the specified criteria

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| Query a| String |  OData query in the format `listDocumentName?queryString`, for example, `Shared Documents?$select=Name&&#36;folderName` |  | x

> | Starting Folder Path a| String |  The starting path of the folder from where to begin the query, relative to the document library selected as part of the query. For example, `/myfolder/level2` searches for folders inside `/document library/myfolder/level2`. NOTE: You must specify the document library as part of the query below. |  / |

Reviewer: The query below? I don't see a query below. Not sure what query this refers to.

| Target Variable a| String |  The name of a variable that stores the operation's output |  |
| Target Value a| String |  An expression that evaluates against the operation's output. The outcome of this expression is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

==== Output
[%autowidth.spread]
|===
|Type |Array of Object
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[resolveObject]]
=== Resolve Object
`<sharepoint:resolve-object>`

Executes an HTTP GET against the SharePoint API. (Executes a POST if the body is blob).

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| Url a| String |  The absolute or relative URL to get |  | x
| Request Type a| Enumeration, one of:

** Get
** Create
** Merge
** Delete |  The request type to be sent. |  Get |

> | Request Body a| Any |  Body of the request to send. The operation sends `null` if empty the request body is empty. |  |

Reviewer: ```suggestion
| Request Body a| Any |  Body of the request to send. The operation sends `null` if the request body is empty. |  |
```

** SECONDS
** MINUTES
** HOURS
** DAYS | A time unit that qualifies the maxIdleTime attribute |  |
|===

[[RedeliveryPolicy]]
=== Redelivery Policy

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Max Redelivery Count a| Number | The maximum number of times a message can be redelivered and processed unsuccessfully before triggering a process-failed message |  |
| Use Secure Hash a| Boolean | Whether to use a secure hash algorithm to identify a redelivered message. |  |
| Message Digest Algorithm a| String | The secure hashing algorithm to use.  | SHA-256 |
| Id Expression a| String | Defines one or more expressions to use to determine when a message has been redelivered. This property can be set only if useSecureHash is false. |  |
| Object Store a| Object Store | The object store where the redelivery counter for each message is stored. |  |
|===

> [[DeletedObject]]

Reviewer: Everything should have descriptions, but this is probably a problem with the source. Let's make a note of all these missing descriptions in our spreadsheet in case it's a bug. If it's not a bug, we need to add all the descriptions in the javadoc.

| Workflow Associations a| Object |  |  |
|===

[[repeatable-in-memory-iterable]]
=== Repeatable In Memory Iterable

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Initial Buffer Size a| Number | The number of instances to initially keep in memory to consume the stream and provide random access to it. If the stream contains more data than can fit into this buffer, then the buffer expands according to the bufferSizeIncrement attribute, with an upper limit of maxInMemorySize. |  100 |
| Buffer Size Increment a| Number | This is by how much the buffer size expands if it exceeds its initial size. Setting a value of zero or lower means that the buffer should not expand, meaning that a STREAM_MAXIMUM_SIZE_EXCEEDED error is raised when the buffer gets full. | 100 |
| Max Buffer Size a| Number | The maximum amount of memory to use. If more memory is used, a STREAM_MAXIMUM_SIZE_EXCEEDED error is raised. A value lower than or equal to zero means no limit. |  |
|===

[[repeatable-file-store-iterable]]
=== Repeatable File Store Iterable

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required

> | In Memory Objects a| Number | The maximum amount of instances to keep in memory. If more than that is required, content on the disk is buffered. |  |

Reviewer: ```suggestion
| In Memory Objects a| Number | The maximum number of instances to keep in memory. If more than what is specified is required, content on the disk is buffered. |  |
```
